Team association

rohit2sharma95 opened the pull request:
Team association is no longer a table so change the logic for user and team association.

- [x] Remove team association model
- [ ] Move team association service to team service

rohit2sharma95 committed:
:fire: Remove team association model

sijusamson committed:
API Updated

Designations changes
Create User changed

sijusamson committed:
Changes for Team Listing

sijusamson committed:
Listing and creatiion API

sijusamson committed:
Team Association removed and changes for team update

sijusamson commented:
- [x]  Move team association service to team service
- [x] Removed Team Association 
- [x] Changes for all other API CRUD.

rohit2sharma95 commented on models/user.js:
```diff
@@ -138,14 +138,17 @@ module.exports = (sequelize, DataTypes) => {
     User.hasMany(models.Leave, { as: 'leaves', foreignKey: 'userId', foreignKeyConstraint: true });
     User.belongsTo(models.Team, { as: 'team', foreignKey: 'teamId', foreignKeyConstraint: true });
-    User.belongsTo(models.Designation, { as: 'designation', foreignKey: 'designation', foreignKeyConstraint: true });
+    User.belongsTo(models.Designation, { as: 'Designation', foreignKey: 'designation', foreignKeyConstraint: true });
```
@sijusamson This will include the _designation_ model as **Designation** instead of **designation** :thinking:

sijusamson replied:
Is it okay If i do like this ?
` User.belongsTo(models.Designation, { as: 'designations', foreignKey: 'designation', foreignKeyConstraint: true });`

rohit2sharma95 replied:
Updated :+1:

rohit2sharma95 commented on src/api/v1/helpers/leaves.helper.js:
```diff
@@ -55,15 +55,15 @@ module.exports = {
          */
-        const teamWithUser = await TeamAssociationService
-          .findTeamOfAUser({ userId: req.userInfo.id });
+        // const teamWithUser = await TeamAssociationService
+        //   .findTeamOfAUser({ userId: req.userInfo.id });
```
Redundant comments

sijusamson replied:
@rohit2sharma95 I didn't removed comments, because earlier we were using TeamAssociation for leave, now as we have removed it we don't need that, but I didn't work on Leave in this branch, I thought it will be better to work for leave in a different branch ?

rohit2sharma95 replied:
Agree :+1:

rohit2sharma95 commented on src/api/v1/helpers/leaves.helper.js:
```diff
@@ -55,15 +55,15 @@ module.exports = {
-        const supervisor = await TeamAssociationService
-          .findSupervisorOfATeam({ teamId: teamWithUser.teamId });
-        usersToBeMailed.push(supervisor.data.email);
+        //  */
+        // const supervisor = await TeamAssociationService
+        //   .findSupervisorOfATeam({ teamId: teamWithUser.teamId });
+        // usersToBeMailed.push(supervisor.data.email);
```
Redundant comments

rohit2sharma95 committed:
Update user model

- Update user model to change the alias for designation
- Replace commented code with TODO flag

rohit2sharma95 committed:
Update model index file as per new version of sequelize
